packages/block-editor/src/components/letter-spacing-control/index.js

```diff
@@ -34,12 +21,16 @@ const CSS_UNITS = [
  * @return {WPElement}                      Letter-spacing control.
  */
 export default function LetterSpacingControl( { value, onChange } ) {
+	const units = useCustomUnits( {
+		availableUnits: useSetting( 'layout.units' ) || [ 'px', 'em', 'rem' ],
```

Comment:
in which situation the fallback is necessary?

Reply:
If for some reason value can't be retrieved from a theme.json file then the fallback will be used. It's a "better safe than sorry" fallback to make sure nothing breaks.

Reply:
It feels we should have a system that provide defaults in the block-editor package for all settings instead of repeating the same code over and over in all places.

Reply:
@nosolosw @jorgefilipecosta how defauts for the block-editor are provided now? (I mean in the frontend without the `core` theme.json)

Reply:
To make sure I follow: do you mean what happens in the editor when a theme doesn't provide a `theme.json`? In the specific case of `layout.units` we don't provide a fallback. We [do](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/use-setting/index.js#L43) for some things like `spacing.units`. We pre-populate the block editor store with [defaults](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/store/defaults.js), but we also don't do that for `__experimentalFeatures`.

Adding the fallback in the `useSetting` hook sounds like something we should explore.

Reply:
Yes, it seems we need default for all the "global" settings. If no setting is provided, we use the default one (kind of like we have the backend with the default theme.json file, maybe we could even remove that one entirely in favor of a frontend defaults that work even in WP-aganostic context)

Reply:
Oh, sorry. Re-reading my comment I don't think I've expressed myself properly. Let me re-write it:

- If themes don't provide their own `theme.json`, the [core](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental-default-theme.json) `theme.json` is used.
- For the specific case of `layout` we don't offer any defaults there. Although, we do have some for `color`, `typography`, etc.

Reply:
The issue with moving this to the client is that mobile won't have access to these. At https://github.com/WordPress/gutenberg/pull/29969 we introduced an endpoint for settings, and we discussed that the block editor could start using the same mechanism to access them, so they're kept in sync. Perhaps that's a good first step?

I also understand how having them in the client may be desirable for a server-agnostic block-editor. Perhaps we can feed the existing core `theme.json` as part of the initialization?

Reply:
Not sure I understand the mobile issue, the mobile should be using the same block-editor package and should have the defaults from the client if they are there.

Reply:
I'm fine if the defaults are duplicated between client and server, if there's a valid reason for it. The endpoint for settings is good to get the settings that are not "defaults" but for the defaults, I don't see the issue for mobile.

Reply:
How do you feel passing defaults as initialization to the block editor?

The fact that we have them in multiple places (client defaults for the store, `useSetting`, server, etc) has been problematic in the past and this issue is also a good demonstration of that.

Reply:
> How do you feel passing defaults as initialization to the block editor?

I think the editor should have defaults bundled, and not require passing settings to work properly.

Reply:
the reason is simple, the block-editor package is independent and it's really not great to force its users to pass a complex settings object to get sane defaults.

Reply:
I've prepared https://github.com/WordPress/gutenberg/pull/32702 that adds a default value for `__experimentalFeatures` in the `@wordpress/block-editor` package. It ports the existing defaults set by the core `theme.json`.